subsys/net/ip/net_context.c

```diff
@@ -626,6 +626,23 @@ int net_context_unref(struct net_context *context)
 
 #if defined(CONFIG_NET_TCP)
 	if (context->tcp) {
+		int i;
+
+		/* Clear the backlog for this TCP context. */
+		for (i = 0; i < CONFIG_NET_TCP_BACKLOG_SIZE; i++) {
```

Comment:
I asked this before, let me ask again: shouldn't we protect access to tcp_backlog[]? Can't the delayed work preempt our code?